micromanager_gui/main_window.py

```diff
@@ -151,34 +103,38 @@ def __init__(
         sig.frameReady.connect(self._on_mda_frame)
 
         # connect buttons
-        self.load_cfg_Button.clicked.connect(self.load_cfg)
-        self.browse_cfg_Button.clicked.connect(self.browse_cfg)
-        self.left_Button.clicked.connect(self.stage_x_left)
-        self.right_Button.clicked.connect(self.stage_x_right)
-        self.y_up_Button.clicked.connect(self.stage_y_up)
-        self.y_down_Button.clicked.connect(self.stage_y_down)
-        self.up_Button.clicked.connect(self.stage_z_up)
-        self.down_Button.clicked.connect(self.stage_z_down)
-
-        self.snap_Button.clicked.connect(self.snap)
-        self.live_Button.clicked.connect(self.toggle_live)
-
-        self.illumination_Button.clicked.connect(self.illumination)
-        self.properties_Button.clicked.connect(self._show_prop_browser)
+        self.cfg.load_cfg_Button.clicked.connect(self.load_cfg)
+        self.cfg.browse_cfg_Button.clicked.connect(self.browse_cfg)
+        self.stages.left_Button.clicked.connect(self.stage_x_left)
+        self.stages.right_Button.clicked.connect(self.stage_x_right)
+        self.stages.y_up_Button.clicked.connect(self.stage_y_up)
+        self.stages.y_down_Button.clicked.connect(self.stage_y_down)
+        self.stages.up_Button.clicked.connect(self.stage_z_up)
+        self.stages.down_Button.clicked.connect(self.stage_z_down)
+
+        self.tab.snap_Button.clicked.connect(self.snap)
+        self.tab.live_Button.clicked.connect(self.toggle_live)
+
+        self.ill.illumination_Button.clicked.connect(self.illumination)
+        self.cfg.properties_Button.clicked.connect(self._show_prop_browser)
+
+        self.stages.focus_device_comboBox.currentTextChanged.connect(
+            self._set_focus_device
+        )
 
         # connect comboBox
-        self.objective_comboBox.currentIndexChanged.connect(self.change_objective)
-        self.bit_comboBox.currentIndexChanged.connect(self.bit_changed)
-        self.bin_comboBox.currentIndexChanged.connect(self.bin_changed)
-        self.snap_channel_comboBox.currentTextChanged.connect(self._channel_changed)
+        self.obj.objective_comboBox.currentIndexChanged.connect(self.change_objective)
+        self.cam.bit_comboBox.currentIndexChanged.connect(self.bit_changed)
+        self.cam.bin_comboBox.currentIndexChanged.connect(self.bin_changed)
+        self.tab.snap_channel_comboBox.currentTextChanged.connect(self._channel_changed)
```

Comment:
I think it makes more sense to move these signal connections (and respective callback methods) into the sub widgets. It feels a bit awkward to have the main window be responsible for managing the state of the sub widgets.

Also this would probably force us to be better about using the `mmc` object as the real source of truth (sorta the model of MV framework) rather than having that work leak into the main window.

Reply:
@ianhi  I think this can be a good idea! However it is maybe better to merge this "new" way to make the gui before doing any further changes...otherwise it might be harder to deal with conflicts. In addition, once this PR is merged, it will be quite easy to merge #107 which can be the new starting point to change widgets signals, etc.... why do you think?